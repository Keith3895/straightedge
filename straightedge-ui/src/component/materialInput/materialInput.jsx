import React from 'react';
import './materialInput.scss';
export class MaterialInput extends React.Component {
    state = {
        value: ""
    };
    constructor(props) {
        super(props);
    }

    handleChange = (e) => {
        this.props.value(e.target.value);
        this.setState({ value: e.target.value });
    }
    render() {
        return (
            <div>
                <div className={`app-div ${this.state.value ? 'is-completed' : ''}`}>
                    <label htmlFor="id" className="app-label" >{this.props.placeHolder}</label>
                    <input data-testid={this.props.uid || new Date().getTime()} className="app-input" value={this.state.value} onChange={this.handleChange} />
                </div>
            </div>
        );
    }
}
