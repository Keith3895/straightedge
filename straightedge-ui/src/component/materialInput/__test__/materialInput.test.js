import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { MaterialInput } from "../materialInput";
import pretty from "pretty";
let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders with or without a name", () => {
  act(() => {
    render(<MaterialInput />, container);
  });
  expect(container.textContent).toBe("");

  act(() => {
    render(<MaterialInput placeHolder="input 1" />, container);
  });
  expect(container.textContent).toBe("input 1");

  act(() => {
    render(<MaterialInput placeHolder="input 2" />, container);
  });
  expect(container.textContent).toBe("input 2");
});

it("renders with placeholder: snapshot", () => {
  act(() => {
    render(<MaterialInput placeHolder='input placeholder'/>, container);
  });
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
