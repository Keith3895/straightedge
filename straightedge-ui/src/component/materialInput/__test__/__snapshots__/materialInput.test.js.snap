// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with placeholder: snapshot 1`] = `
"<div>
  <div class=\\"app-div \\"><label for=\\"id\\" class=\\"app-label\\">input placeholder</label><input data-testid=\\"1590585791205\\" class=\\"app-input\\" value=\\"\\"></div>
</div>"
`;
